### ISSUE TITLE
zmodule hardcodes `master` as the default branch rather than using the repository's default branch when no branch name is specified

### ISSUE THREAD

**Issue by cowboyd** [zimfw/zimfw#422]
```
$ zimfw info
zimfw version: 1.3.2 (previous commit is ccace0c)
ZIM_HOME:      /Users/cowboyd/.zim
Zsh version:   5.4.2
System info:   Darwin blackfyre.local 20.1.0 Darwin Kernel Version 20.1.0: Sat Oct 31 00:07:10 PDT 2020; root:xnu-7195.50.7~2/RELEASE_ARM64_T8101 x86_64
```

Description
-----------
It's becoming more and more common for developers to [rename their default branch away from `master`][1] which means that it will be come more and more likely for `zimfw install` to break if a module's  branch is always assumed to be `master`. 

If no branch is specified then, `zimfw install` should just clone the repo without specifying a branch at all thereby pulling the repository's default branch.

Steps to reproduce
------------------
I ran into this while building [a plugin for volta][2], and following the advice to [use a meaningful name for the  default branch][3]

When I tried this to my `~/.zimrc`

```
zmodule cowboyd/zimfw-volta
```

a `zimfw intall` resulted in the following error message:

``` shell
➜ ~ zimfw install
x zimfw-volta: Error during git clone
  fatal: Remote branch master not found in upstream origin
```

I'd be happy to try and implement a fix if anybody can point me in the right direction

[1]: https://github.com/github/renaming
[2]: https://github.com/cowboyd/zimfw-volta
[3]: https://frontside.com/blog/2020-12-11-default-branch-names/

**Comment by ericbn** [comment]
Good point! I'll work on it.

**Comment by ericbn** [comment]
Just released this with version 1.4.0.
